src/resources/js/accordion.js

```diff
@@ -1,57 +1,97 @@
-( function( window, $ ) {
-	var accordions = document.getElementsByClassName( 'accordion' );
-
-	var accordion_index = ( undefined === accordion_index ) ? 0 : accordion_index++;
-
-	for ( var i = 0, aLen = accordions.length; i < aLen; i++ ) {
-		var accordion = accordions[i];
-		var accordion_content = accordion.getElementsByClassName( 'accordion-content' );
-		var accordion_headers  = accordion.getElementsByClassName( 'accordion-header' );
-
-			for ( var t = 0, hLen = accordion_headers.length; t < hLen; t++ ) {
-				var header = accordion_headers[t]
-				header.setAttribute( 'id', 'tab' + accordion_index + '-' + t );
-				header.setAttribute( 'aria-selected', 'false' );
-				header.setAttribute( 'aria-expanded', 'false' );
-				header.setAttribute( 'aria-controls', 'panel' + accordion_index + '-' + t );
-				header.setAttribute( 'role', 'tab' );
-
-				header.addEventListener( 'click', handle_accordion );
-
-				/**
-				 * Handles the changes (both visual and aria) for the accordion clicks
-				 *
-				 * @since TBD
-				 */
-				function handle_accordion() {
-					var next_panel = header.nextElementSibling,
-					next_panel_label = next_panel.querySelector( '.accordion-label' );
-
-					header.classList.toggle( 'is-active' );
-					next_panel.classList.toggle( 'is-active' );
-					next_panel_label.setAttribute( 'tabindex', -1 );
-					next_panel_label.focus();
-
-
-					if ( next_panel.classList.contains( 'is-active' ) ) {
-						header.setAttribute( 'aria-selected', 'true' );
-						header.setAttribute( 'aria-expanded', 'true' );
-						next_panel.setAttribute( 'aria-hidden', 'false' );
-					} else {
-						header.setAttribute( 'aria-selected', 'false' );
-						header.setAttribute( 'aria-expanded', 'false' );
-						next_panel.setAttribute( 'aria-hidden', 'true' );
-					}
-				}
+( function() {
+	'use strict';
+
+	// This is our global accordion index to keep unique ids
+	var topIndex = 0;
+
+	var MT_accordion = function( options, callback ) {
+		if ( 'undefined' === typeof options.target ) {
+			return false;
+		}
+
+		var accordion = document.querySelector( options.target );
+
+		if ( ! accordion ) {
+			return;
+		}
+
+		// Simple iterator for reuse
+		var forEach = function( array, callback, scope ) {
+			for ( var i = 0, imax = array.length; i < imax; i++ ) {
+				callback.call( scope, i, array[i] ); // passes back stuff we need
 			}
+		};
+
+		var accordionContent = accordion.getElementsByClassName( 'accordion-content' ),
+			accordionHeader  = accordion.getElementsByClassName( 'accordion-header' );
```

Comment:
Consider multiple `var` statements instead of a single multiline `var`, per notes in our engineering standards?